uniswap-v2-core/test/UniswapV2Pair.spec.ts

```diff
@@ -437,4 +439,62 @@ describe('UniswapV2Pair', () => {
     expect(await token0.balanceOf(pair.address), "Token 0 balance of pair").to.eq(bigNumberify(1000).add(token0ExpBalVexchange))
     expect(await token1.balanceOf(pair.address), "Token 1 balance of pair").to.eq(bigNumberify(1000).add(token1ExpBalVexchange))
   })
+
+
+  /**
+   *  recoverToken - error handling for invalid tokens 
+   */
+  it('recoverToken:invalidToken', async () => {
+    await expect(pair.recoverToken(token0.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
+    await expect(pair.recoverToken(token1.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
+  
+    // address is a random invalid token address
+    await expect(pair.recoverToken("0x3704E657053C02411aA2Fd0599e75C3d817F81BC")).to.be.reverted
```

Comment:
Would this revert with `UniswapV2: TRANSFER_FAILED`?

Reply:
Bringing @OliverNChalk' s comment above here, style wise, although it is commented making it clear, it's probably better to make this a constant with a clear name (INVALID_ADDRESS etc). 

And yes please add to specific revert message expected using revertedWith().  I think it will be "Vexchange: RECOVERER_ZERO_ADDRESS"

Reply:
In my test setup the error I got for the invalid token address case (bottom one), was 
`RuntimeError: VM Exception while processing transaction: revert` 
So it's unlikely that it even reached line 108 of `UniswapV2Pair.sol` . But it's unclear at which line it failed. It could be line 262 in or it could be in line 107. Is there any way for me to test this? I would need to import the IERC20 interface into the test file and call the balanceOf function? 

also I noticed for the invalid token address test case that I forgot to set the recoverer's address in the first place. Cuz if the recoverer's address is not even set, the 
`require(recoverer != address(0), "Vexchange: RECOVERER_ZERO_ADDRESS");` 
would already fail before we even reach `_safeTransfer`. So I'll add in the code to set the recovery address.

Reply:
Yeah, I'm not sure what the first error will be if we call a non-existant contract. I.e. if it's a valid address, then EVM will assume its EOA and thus my throw because we tried to call an EOA like a contract?

I'm okay not to invest a lot of time solving this issue because we would expect grabage in garbage out here